File: Arrays/Algorithms/Sliding_Window/Variable_Size/longest_subarray_sum_at_most_k.cpp
#include <bits/stdc++.h>
using namespace std;

int longestSubarraySumAtMostK(const vector<int>& arr, int k) {
	int l = 0, sum = 0, res = 0;
	for (int r = 0; r < arr.size(); r++) {
		sum += arr[r];
		while (sum > k && l <= r) {
			sum -= arr[l++];
		}
		res = max(res, r - l + 1);
	}
	return res;
}

int main() {
	int n, k;
	cin >> n >> k;
	vector<int> arr(n);
	for (int i = 0; i < n; i++) cin >> arr[i];
	cout << longestSubarraySumAtMostK(arr, k) << endl;
	return 0;
}

File: Arrays/Algorithms/Sliding_Window/Variable_Size/longest_subarray_sum_at_most_k.py
def longest_subarray_sum_at_most_k(arr, k):
	l = s = res = 0
	for r in range(len(arr)):
		s += arr[r]
		while s > k and l <= r:
			s -= arr[l]
			l += 1
		res = max(res, r - l + 1)
	return res

if __name__ == "__main__":
	n, k = map(int, input().split())
	arr = list(map(int, input().split()))
	print(longest_subarray_sum_at_most_k(arr, k))
